specs-go/v1/config.go

```diff
@@ -75,6 +75,7 @@ type History struct {
 }
 
 // Image is the JSON structure which describes some basic information about the image.
+// This provides the `application/vnd.oci.image.config.v1+json` mediatype when marshalled to JSON.
```

Comment:
@stevvooe woah. The `mediatype` is as necessary as a document providing the version or schema that it represents. Why have any of them, if this is the case?
Regardless, this leaves a mess for implementations to require a big case statement for reflect and discovery what kind of document and version is even being dealt with.